File: Math/789. Escape The Ghosts/789. Escape The Ghosts.java
class Solution {
    public boolean escapeGhosts(int[][] ghosts, int[] target) {
        int dis = Math.abs(target[0]) + Math.abs(target[1]);
        for (int[] g : ghosts) {
            int gd = Math.abs(target[0]-g[0]) + Math.abs(target[1]-g[1]);
            if (gd <= dis) return false;
        }
        return true;
    }
}

File: Math/789. Escape The Ghosts/789. Escape The Ghosts.cpp
class Solution {
public:
    bool escapeGhosts(vector<vector<int>>& ghosts, vector<int>& target) {
        int dis = abs(target[0]) + abs(target[1]);
        for (auto g : ghosts) {
            int gd = abs(target[0]-g[0]) + abs(target[1]-g[1]);
            if (gd <= dis) return false;
        }
        return true;
    }
};
